File: AssetManagementSystem-Client/src/app/models/asset.ts
export interface SoftwareLicense {
    licenseName: string;
}

export interface Asset {
    assetId: string;
    name: string;
    serialNumber: string;
    statusId: number;
    employeeName?: string;
    warrantyName?: string;
    softwareLicenses?: SoftwareLicense[];
}

export interface AssigneeAsset{
    assetId: string;
    employeeId: string;
}

export interface AddedAsset{
    name: string;
    serialNumber: string;
    warrantyName: string;
}

export interface EditAsset{
    name: string;
    serialNumber?: string;
    warrantyName?: string;
    employeeName?: string;
}

File: AssetManagementSystem/Models/Asset.cs
using System.ComponentModel.DataAnnotations;

namespace AssetManagementSystem.Models
{
    public class Asset
    {
        [Key]
        public int Id { get; set; }
        public string AssetId { get; set; }
        public string Name { get; set; }
        public string SerialNumber { get; set; }

        public int StatusId { get; set; } // ForeignKey
        public Status Status { get; set; }

        public WarrantyCard WarrantyCard { get; set; }

        public int? EmployeeId { get; set; } // ForeignKey

        public Employee Employee { get; set; }

        public ICollection<AssetSoftware> AssetSoftwares{ get; set; }

    }
}
